Add information about selectors for querySelector*

dorward opened the pull request:
<!-- Please provide the following information to help us review this PR: -->

> What was wrong/why is this fix needed? (quick summary only)

The page lacked information about what selectors could be used.

> MDN URL of the main page changed

https://developer.mozilla.org/en-US/docs/Web/API/Document_object_model/Locating_DOM_elements_using_selectors

> Issue number (if there is an associated issue)

#5503 

> Anything else that could help us review it

dorward committed:
#5503 : Add information about selectors for querySelector*

Adds information about selectors for querySelector and querySelectorAll

* Adds links to the selectors reference on MDN
* Updates the W3C reference to the current specification (previous was marked as superceeded)
* Adds a description of which selectors can be used
* Moves examples to their own section
* Changes a reference to “CSS class” to just “class”. The class attribute is defined by HTML and not CSS.

wbamberg commented:
Thanks for filing this PR, @dorward ! Would you mind please submitting a version that doesn't add hard-wrapping to the lines of this file? It makes it much harder to see what changed.

dorward committed:
#5503 : Fix line breaks, correct use of “either” and classes.

This rolls back the last commit and reapplies the changes to remove the auto formatting applied by my editor.

It also tweaks some language around classes since elements have a list of classes, not a single one.

dorward commented:
> Thanks for filing this PR, @dorward ! Would you mind please submitting a version that doesn't add hard-wrapping to the lines of this file? It makes it much harder to see what changed.

Sorry about that. I had my editor set to soft wrapping and didn't notice it applying hard wrapping with autoformat on save. 🤦🏻‍♂️ (At least until I looked at the diff, by which time I was ready to hit the sack!)

dorward committed:
#5503 : Restore white space at end of document

sideshowbarker commented on files/en-us/web/api/document_object_model/locating_dom_elements_using_selectors/index.html:
```diff
@@ -35,12 +45,11 @@ <h2 id="Selectors">Selectors</h2>
 <ul>
- <li><a href="https://www.w3.org/TR/selectors-api/">Selectors API</a></li>
+ <li><a href="https://www.w3.org/TR/selectors-3/">Selectors Level 3 (W3C Recommendation)</a></li>
+ <li>{{cssxref("CSS_Selectors", "CSS Selectors")}}</li>
```
```suggestion
 <li><a href="/en-US/docs/Web/CSS/CSS_Selectors">CSS Selectors</a></li>
```
…because the `cssxref` macro otherwise causes it to get rendered as `css selectors` (lowercase, monospace).

In general, anyway, it’s always OK to not use the macros but instead put in a normal hyperlink.

In fact, these days the devs working on the MDN infrastructure/tooling strongly prefer that we use normal hyperlinks rather than macros (in part I guess because of needing to deal with inquiries/confusion about the macros doing stuff that some people (like me) don’t expect them to do.

(I can imagine why the `cssxref` does this automatic lowercase+monospace things, but I guess I don’t reference the CSS docs often enough to have run into before myself. Or at least I don’t remember…)

sideshowbarker reviewed (changes requested):
This looks beautiful to me modulo the few nits/quirks I added commit suggestions for.

sideshowbarker commented:
> Sorry about that. I had my editor set to soft wrapping and didn't notice it applying hard wrapping with autoformat on save.

This reminds me that myself, working with the HTML sources after they had been moved out from the wiki and over to github, I didn’t much like that most of the lines in the sources weren’t hard-wrapped at 80 columns or whatever to begin with. I didn’t have my editor of choice, vim, configured to deal well with long-lines editing. 

But after spending some time getting my vim set up to handle it well, I’ve now personally kind of flipped around 180 degrees on it, and very much prefer working with the sources with no hard wrapping  (except of course at the paragraph breaks and all).

sideshowbarker committed:
Update files/en-us/web/api/document_object_model/locating_dom_elements_using_selectors/index.html

sideshowbarker committed:
Update files/en-us/web/api/document_object_model/locating_dom_elements_using_selectors/index.html

sideshowbarker committed:
Update files/en-us/web/api/document_object_model/locating_dom_elements_using_selectors/index.html
